routing/leaps_postprocessor.cpp

```diff
@@ -41,16 +41,58 @@ namespace routing
 // static
 size_t const LeapsPostProcessor::kMaxStep = 5;
 
-LeapsPostProcessor::LeapsPostProcessor(std::vector<Segment> const & path, IndexGraphStarter & starter)
-  : m_path(path),
-    m_starter(starter),
-    m_bfs(starter),
-    m_prefixSumETA(path.size(), 0.0)
+LeapsPostProcessor::LeapsPostProcessor(std::vector<Segment> const & path,
+                                       IndexGraphStarter & starter)
+  : m_starter(starter),
+    m_bfs(starter)
 {
-  for (size_t i = 1; i < path.size(); ++i)
+  std::map<size_t, size_t> jumps;
+  std::map<Segment, size_t> segmentToIndex;
+
```

Comment:
Может нам написать, зачем мы делаем эту фильтрацию и от чего хотим защититься? Т.е. в чем причина и основную идею.

Reply:
```cpp
/// \brief It is assumed that there are no cycles in path.
/// But sometimes it's wrong (because of heuristic searching of intermediate
/// points (mwms' exists) in LeapsOnly routing algorithm). 
/// Look here for screenshots of such problem: https://github.com/mapsme/omim/pull/11091
/// This code removes cycles from path before |Init()|.
/// The algorithm saves jumps between the same segments, so if we have such loop:
/// s1, s2, ... , s10, s1, s11, s12, ... , s100, where "s3" means segment number 3.
/// ^------------------^ <--- loop (from s1 to s1).
/// Thus if we look at the first s1, we will find out that exists a jump to second s1
/// and we will skip all segments between first and second s1 and the result path will be:
/// s1, s11, s12, ... , s100

```

Reply:
I wrote some minor corrections.